### PR TITLE
Query API CORS Support

### PR DESCRIPTION
**Pull request by erans**
Controlled from an environment variable named "REDASH_QUERIES_RESULT_CORS". Can contain a comma separated list of domains.

### CODE REVIEW

**Comment by arikfr** [comment]
It includes your Python runner changes even though I merged it already. Can you rebase, just to make sure we're not overriding anything?

File: redash/controllers.py [hunk 1]
```diff
@@ -488,6 +488,22 @@ def csv_response(query_result):
         headers.update(cache_headers)
         return make_response(s.getvalue(), 200, headers)
 
+    @staticmethod
+    def add_access_control_allow_origin_header(headers={}):
```
Comment (arikfr) [inline redash/controllers.py]: I'm not sure about Flask's life cycle, but if controllers are reused, the default headers will be reused, aren't they?

File: redash/controllers.py [hunk 2]
```diff
@@ -488,6 +488,22 @@ def csv_response(query_result):
...
+            if origin in settings.QUERIES_RESULT_CORS:
+                headers['Access-Control-Allow-Origin'] = origin
+                headers['Access-Control-Allow-Credentials'] = 'true'
+                if request.method == 'OPTIONS':
+                    headers['Access-Control-Request-Method'] = 'GET, POST, PUT'
+                    headers['Access-Control-Allow-Headers'] = 'Content-Type'
+
+    @require_permission('view_query')
+    def options(self, query_id=None, query_result_id=None, filetype='json'):
+        self.add_access_control_allow_origin_header()
```
Comment (arikfr) [inline redash/controllers.py]: If you don't pass anything, it just sets the local headers variables. How is it working then?

File: redash/controllers.py [hunk 3]
```diff
@@ -517,9 +533,15 @@ def get(self, query_id=None, query_result_id=None, filetype='json'):
 
                 record_event.delay(event)
 
+            headers = {}
+
+            if settings.QUERIES_RESULT_CORS and len(settings.QUERIES_RESULT_CORS) > 0:
```
Comment (arikfr) [inline redash/controllers.py]: `settings.QUERIES_RESULTS_CORS` won't be truthy if len not > 0...

File: redash/controllers.py (hunk 1 above)
Comment (erans) [inline redash/controllers.py]: they should be. I think the generate make_response adds the headers you send on top of the default headers.

File: redash/controllers.py (hunk 2 above)
Comment (erans) [inline redash/controllers.py]: true. It was from a previous version. I'll remove the default.

File: redash/controllers.py (hunk 3 above)
Comment (erans) [inline redash/controllers.py]: yeah, I think I can remove that and keep the len.

**Pull request closed without merging**